numpy/core/tests/test_shape_base.py

```diff
@@ -330,5 +333,121 @@ def test_stack():
                         stack, [m, m])
 
 
+class TestBlock(TestCase):
+    def test_block_simple_row_wise(self):
+        A = np.ones((2, 2))
+        B = 2 * A
+        desired = np.array([[1, 1, 2, 2],
+                            [1, 1, 2, 2]])
+        result = block([A, B])
+        assert_equal(desired, result)
+        # with tuples
+        result = block((A, B))
+        assert_equal(desired, result)
+
+    def test_block_simple_column_wise(self):
+        A = np.ones((2, 2))
+        B = 2 * A
+        expected = np.array([[1, 1],
+                             [1, 1],
+                             [2, 2],
+                             [2, 2]])
+        result = block([[A], [B]])
+        assert_equal(expected, result)
+
+    def test_block_with_1d_arrays_row_wise(self):
+        # # # 1-D vectors are treated as row arrays
+        a = np.array([1, 2, 3])
+        b = np.array([2, 3, 4])
+        expected = np.array([1, 2, 3, 2, 3, 4])
+        result = block([a, b])
+        assert_equal(expected, result)
+
+    def test_block_with_1d_arrays_multiple_rows(self):
+        a = np.array([1, 2, 3])
+        b = np.array([2, 3, 4])
+        expected = np.array([[1, 2, 3, 2, 3, 4],
+                             [1, 2, 3, 2, 3, 4]])
+        result = block([[a, b], [a, b]])
+        assert_equal(expected, result)
+
+    def test_block_with_1d_arrays_column_wise(self):
+        # # # 1-D vectors are treated as row arrays
+        a = np.array([1, 2, 3])
+        b = np.array([2, 3, 4])
+        expected = np.array([[1, 2, 3],
+                             [2, 3, 4]])
+        result = block([[a], [b]])
+        assert_equal(expected, result)
+
+    def test_block_mixed_1d_and_2d(self):
+        A = np.ones((2, 2))
```

Comment:
I've renamed these to `a_2d`. The fact that I had to justify the capitals suggests that they failed at conveying the meaning they intended, so let's go with something more obvious